tests/e2e/support/objects/app-files/resource/actions.ts

```diff
@@ -684,6 +689,114 @@ export const pasteResource = async (
+    for (const resource of resources) {
+      waitForMoveResponses.push(
+        page.waitForResponse(
+          (resp) =>
+            resp.url().endsWith(resource) &&
+            resp.status() === 201 &&
+            resp.request().method() === 'MOVE'
+        )
+      )
+    }
+  }
+
+  switch (method) {
+    case 'dropdown-menu': {
+      // after selecting multiple resources, resources can be copied or moved by clicking on any of the selected resources
+      await page.locator(highlightedFileRowSelector).first().click({ button: 'right' })
+      await page.locator(util.format(filesContextMenuAction, action)).click()
+
+      await page.locator(breadcrumbRoot).click()
+      const newLocationPath = newLocation.split('/')
+      for (const path of newLocationPath) {
+        if (path !== 'Personal') {
+          await clickResource({ page, path: path })
+        }
+      }
+      await page.locator(filesView).click({ button: 'right' })
+      await page.locator(util.format(filesContextMenuAction, 'copy')).click()
+      await waitForResources({
+        page,
+        names: resources
+      })
+      break
+    }
+    case 'batch-action': {
+      await page.locator(util.format(filesBatchAction, action)).click()
+
+      await page.locator(breadcrumbRoot).click()
+      const newLocationPath = newLocation.split('/')
+      for (const path of newLocationPath) {
+        if (path !== 'Personal') {
+          await clickResource({ page, path: path })
+        }
+      }
+      await page.locator(pasteButton).click()
+      await waitForResources({
+        page,
+        names: resources
+      })
+
+      break
+    }
+    case 'keyboard': {
+      const keyValue = action === 'copy' ? 'c' : 'x'
+      await page.keyboard.press(`Control+${keyValue}`)
+      await page.locator(breadcrumbRoot).click()
+      const newLocationPath = newLocation.split('/')
+      for (const path of newLocationPath) {
+        if (path !== 'Personal') {
+          await clickResource({ page, path: path })
+        }
+      }
+      await page.keyboard.press('Control+v')
+      await waitForResources({
+        page,
+        names: resources
+      })
+      break
+    }
+    case 'drag-drop': {
+      const source = page.locator(highlightedFileRowSelector).first()
+      const target = page.locator(util.format(resourceNameSelector, newLocation))
+
+      await Promise.all([...waitForMoveResponses, source.dragTo(target)])
+
+      await target.click()
+      await waitForResources({
+        page,
+        names: resources
+      })
+      break
+    }
+    case 'drag-drop-breadcrumb': {
+      const source = page.locator(highlightedFileRowSelector).first()
+      const target = page.locator(util.format(breadcrumbResourceNameSelector, newLocation))
+
+      await Promise.all([...waitForMoveResponses, source.dragTo(target)])
+
+      await target.click()
+      await waitForResources({
+        page,
+        names: resources
+      })
+      break
+    }
+  }
```

Comment:
Then now we can wait for resources at the last
```suggestion
    }
    await waitForResources({
        page,
        names: resources
      })
  }
```